### ISSUE TITLE
False warning of property hiding inherited method member

### ISSUE THREAD

**Issue by AlFasGD** [dotnet/roslyn#63509]
**Version Used**: 
- 17.4.0 Preview 1.0
- `main` branch @ 54aeea4 (in SharpLab)

**Steps to Reproduce**:

1. Define a class C with a method M (no constraint on parameter count)
2. Define a class D that inherits C and defines a property M that does not return a delegate (verified that the delegate property will properly hide the base method)

[SharpLab repro](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK4wD4DcLIAQAmeAvHjAKYDueAIgBQCUA3ALABQsCeYJhAdAFlWHGFzA9SBQXQCMcPACZmeAPQq8CABblu5LQHsiUAM5l9XCAGMEqCABs7ATzyaoBAuRjt2AAQDMeD4KeADC7ADe7HjRgQGceAJ08RDy8cAMUTGRbDG5gQDseBB4ANR4wMK5AL7sNWz+gcE0eCChEZnRDRTU8QIkAHwJdH7ySsJVQA)

**Expected Behavior**:
No warning; D.M is a property and C.M is a method, so no hiding

**Actual Behavior**:
A CS0108 warning is reported saying that D.M hides inherited member C.M, despite both being normally callable and not hidden.
Also, adding the `new` keyword to D.M's declaration will not show a CS0109 warning (redundant `new` keyword), as it is considered as a hiding declaration.

**Comment by Youssef1313** [comment]
I think the property indeed hides the method:

```csharp
using System;

public class C
{
    public int M(int a, int b)
    {
        return a + b;
    }
}
public class D : C
{
    public int M => M(3, 2);

    public void F() => Call(M, 0, 0); // The property hides base method M here.

    private int Call(Func<int, int, int> x, int a, int b) => x(a, b);
}
```

Can someone from compiler team confirm?

**Comment by AlFasGD** [comment]
As it's being interpreted as a delegate, it is indeed hidden. That's the scenario where hiding is performed. In simple invocation scenarios, the method is not hidden, and the warning appears wrong/misleading. Based on that, I believe it would be better to introduce a new warning that specifies the hiding scope (when used as a delegate), and applying the normal hiding warning only when the property is of a delegate type.

**Comment by jaredpar** [comment]
This behavior is "By Design". As noted this does hide the member. Pretty much every warning has ways it could be improved for specific cases. Do not think this one meets the bar though. It's been this way since practically C# 1.0 and we've gotten very little feedback on it.
